crates/core/tedge_agent/src/software_manager/actor.rs

```diff
@@ -74,26 +99,26 @@ impl Actor for SoftwareManagerActor {
 
         self.process_pending_sm_operation().await?;
 
-        while let Some(request) = self.message_box.recv().await {
-            match request {
-                SoftwareRequest::SoftwareUpdateRequest(request) => {
-                    if let Err(err) = self
-                        .handle_software_update_operation(&request, &mut plugins, &operation_logs)
-                        .await
-                    {
-                        error!("{:?}", err);
-                    }
-                }
-                SoftwareRequest::SoftwareListRequest(request) => {
-                    if let Err(err) = self
-                        .handle_software_list_operation(&request, &plugins, &operation_logs)
-                        .await
-                    {
-                        error!("{:?}", err);
-                    }
+        let mut input_receiver = self.input_receiver.take().ok_or(RuntimeError::ActorError(
+            anyhow::anyhow!("actor can't be run more than once").into(),
+        ))?;
```

Comment:
Do I understand correctly?

You take the receiver out of `self` to be able to `tokio::select` on `self.handle_request()` and `self.input_receiver.recv_signal()`, two methods that expect `&mut self`. Am I correct?

Reply:
That's exactly right.

Reply:
Thanks